pkg/services/serviceaccounts/database/database.go

```diff
@@ -247,6 +247,35 @@ func (s *ServiceAccountsStoreImpl) UpdateServiceAccount(ctx context.Context,
 	}, err
 }
 
+func (s *ServiceAccountsStoreImpl) SearchOrgServiceAccounts(ctx context.Context, query *models.SearchOrgUsersQuery) ([]*serviceaccounts.ServiceAccountDTO, error) {
+	if !query.IsServiceAccount {
+		return nil, fmt.Errorf("invalid query for service accounts")
+	}
+	// translate between users and serviceaccountsDTO
+	err := s.sqlStore.SearchOrgUsers(ctx, query)
```

Comment:
We'll need to extract this function into this package (or this function even) for AC reasons.

https://github.com/grafana/grafana/blob/ee83bceb3a493be358825584df84bbc01b4d5b57/pkg/services/sqlstore/org_users.go#L186-L193

is used to filter based on permissions on "users:read", we're using "service_accounts:read"

You can then drop the scope:all from the registration as the service account store will handle the filtering

https://github.com/grafana/grafana/blob/ee83bceb3a493be358825584df84bbc01b4d5b57/pkg/api/api.go#L217